src/bootstrap/test.rs

```diff
@@ -144,6 +144,47 @@ You can skip linkcheck with --exclude src/tools/linkchecker"
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub struct HtmlCheck {
+    target: TargetSelection,
+}
+
+impl Step for HtmlCheck {
+    type Output = ();
+    const DEFAULT: bool = true;
+    const ONLY_HOSTS: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.path("src/tools/html-checker")
+    }
+
+    fn make_run(run: RunConfig<'_>) {
+        run.builder.ensure(HtmlCheck { target: run.target });
+    }
+
+    fn run(self, builder: &Builder<'_>) {
+        if !Command::new("tidy")
+            .arg("--version")
+            .stdout(Stdio::null())
+            .status()
+            .map_or(false, |status| status.success())
+        {
+            eprintln!("not running HTML-check tool because `tidy` is missing");
+            return;
+        }
+        // Ensure that a few different kinds of documentation are available.
+        builder.ensure(crate::doc::Std { target: self.target, stage: builder.top_stage });
+        builder.ensure(crate::doc::Rustc { target: self.target, stage: builder.top_stage });
```

Comment:
Usually the way we do this is we invoke `builder.default_doc(&[])`. I'd copy whatever the linkcheck step does before running linkcheck.